### DISCUSSION TITLE
URI expiration or some workaround

### DISCUSSION THREAD

**Discussion by johnpaulcas** [WalletConnect/walletconnect-monorepo#3185] [Developer Support] [answered]
We're developing a backend API for an Unreal Engine game due to the lack of an existing SDK. At the moment, we're using the Sign Client SDK, and are specifically dealing with the code: ```const { uri, approval } = await this.signClient.connect({ requiredNamespaces });.```

Our challenge revolves around the approval method on our backend. Once the URI is returned, we wait for the user's approval to establish the session. Currently, we've set the approval timeout to roughly one minute.

Our question is: What's the most effective way to manage this situation? More specifically, is there an optimal method for setting the URI to expire after a specified number of minutes or cancelling it outright?

**Comment by glitch-txs** [comment]
Hi @ganchoradkov could you help here please 🙏🙂

**Comment by johnpaulcas** [comment]
Hello @glitch-txs and @ganchoradkov, I'm reaching out to follow up on the previously discussed matter. thanks!

**Comment by glitch-txs** [comment, marked as answer]
Hi! you can await the approval method which is going to resolve with the session object once approved by the wallet, or it will throw if rejected, also the URI expires in 5 minutes if its not used as well as the proposal.

**Comment by W3stside** [comment]
hi @glitch-txs is there a way to extend this hard-coded 5 minutes expiry time in Web3Modal (in my case web sdk for react)? e.g we have clients connecting Safe Wallet with multiple signers via WalletConnect QR code and the tx expires as the other remote signers sometimes need more than 5 min to sign. Any ideas?

**Reply by glitch-txs** [reply]
Not possible atm, but feel free to create an issue as a feature request

**Reply by W3stside** [reply]
I'd be happy to create a pr for v2, if possible (seeing as you're now V3)

**Reply by glitch-txs** [reply]
You'll need to go more low level though, in the monorepo, feel free to ping Gancho there
